iphone/Classes/TiUIShortcutProxy.m

```diff
@@ -0,0 +1,156 @@
+    [shortcutsToReturn addObject:[[[TiUIShortcutItemProxy alloc] initWithShortcutItem:item] autorelease]];
+  }
+
+  return shortcutsToReturn;
+}
+
+- (NSArray<TiUIShortcutItemProxy *> *)staticItems
+{
+  NSMutableArray<TiUIShortcutItemProxy *> *shortcutsToReturn = [NSMutableArray array];
+  NSArray *shortcuts = [NSBundle mainBundle].infoDictionary[@"UIApplicationShortcutItems"];
+
+  if (shortcuts == nil || [shortcuts count] == 0) {
+    return @[];
+  }
+
+  for (id item in shortcuts) {
+    // We need to map the plist-keys manually for static shortcuts
+    NSString *type = [item valueForKey:@"UIApplicationShortcutItemType"];
+    NSString *title = [item valueForKey:@"UIApplicationShortcutItemTitle"];
+    NSString *subtitle = [item valueForKey:@"UIApplicationShortcutItemSubtitle"];
+    UIApplicationShortcutIcon *icon = [UIApplicationShortcutIcon iconWithType:[TiUtils intValue:[item valueForKey:@"UIApplicationShortcutItemIconType"]]];
+    NSDictionary *userInfo = [item valueForKey:@"UIApplicationShortcutItemUserInfo"];
+
+    UIApplicationShortcutItem *shortcut = [[[UIApplicationShortcutItem alloc] initWithType:type
+                                                                            localizedTitle:title
+                                                                         localizedSubtitle:subtitle
+                                                                                      icon:icon
+                                                                                  userInfo:userInfo] autorelease];
+
+    [shortcutsToReturn addObject:[[[TiUIShortcutItemProxy alloc] initWithShortcutItem:shortcut] autorelease]];
+  }
+
+  return shortcutsToReturn;
+}
+
+- (TiUIShortcutItemProxy *)getById:(NSString *)identifier
+{
+  NSArray<UIApplicationShortcutItem *> *shortcuts = [UIApplication sharedApplication].shortcutItems;
+  for (UIApplicationShortcutItem *item in shortcuts) {
+    if ([item.type isEqualToString:[TiUtils stringValue:identifier]]) {
+      return [[[TiUIShortcutItemProxy alloc] initWithShortcutItem:item] autorelease];
+    }
+  }
+
+  return nil;
+}
+
+- (void)remove:(TiUIShortcutItemProxy *)shortcut
+{
+  NSString *key = [shortcut shortcutItem].type;
+
+  NSMutableArray<UIApplicationShortcutItem *> *shortcuts = (NSMutableArray<UIApplicationShortcutItem *> *)[UIApplication sharedApplication].shortcutItems;
+  for (UIApplicationShortcutItem *item in shortcuts) {
+    if ([item.type isEqualToString:[shortcut shortcutItem].type]) {
+      [shortcuts removeObject:item];
+      break;
+    }
+  }
+  [UIApplication sharedApplication].shortcutItems = shortcuts;
+}
+
+- (void)removeAll
+{
+  [UIApplication sharedApplication].shortcutItems = nil;
+}
+
+- (void)add:(TiUIShortcutItemProxy *)shortcut
+{
+  NSMutableArray<UIApplicationShortcutItem *> *shortcuts = (NSMutableArray<UIApplicationShortcutItem *> *)[UIApplication sharedApplication].shortcutItems;
+
+  // remove the previous shortcutitem of same id if exists
```

Comment:
do we need to worry about position? In other words, should we replace the existing shortcut with that id so we retain the same index/position in the array? or does position not matter here?

Reply:
Good catch! Position matters. Updated to maintain position.